conda_smithy/schema.py

```diff
@@ -0,0 +1,1172 @@
+    )
+
+    settings_win: AzureRunnerSettings = Field(
+        default_factory=lambda: AzureRunnerSettings(
+            pool={"vmImage": "windows-2022"},
+            variables={
+                "CONDA_BLD_PATH": "D:\\\\bld\\\\",
+                "UPLOAD_TEMP": "D:\\\\tmp",
+            },
+        ),
+        description="Windows-specific settings for self-hosted runners",
+    )
+
+    user_or_org: Optional[Union[str, Nullable]] = Field(
+        default=None,
+        description="The name of the GitHub user or organization, if passed with \
+        the GithubConfig provider, must comply with the value of the user_or_org field",
+        exclude=True,  # Will not be rendered in the model dump
+    )
+
+    store_build_artifacts: Optional[bool] = Field(
+        default=False,
+        description="Store the conda build_artifacts directory as an \
+        Azure pipeline artifact",
+    )
+
+    timeout_minutes: Optional[Union[int, Nullable]] = Field(
+        default=None,
+        description="The maximum amount of time (in minutes) that a \
+            job can run before it is automatically canceled",
+    )
+
+
+class GithubConfig(BaseModel):
+    user_or_org: Optional[str] = Field(
+        description="The name of the GitHub user or organization, \
+        if passed with the AzureConfig provider, must comply with the value of the user_or_org field",
+        default="conda-forge",
+    )
+    repo_name: Optional[str] = Field(
+        description="The name of the repository",
+        default="",
+    )
+    branch_name: Optional[str] = Field(
+        description="The name of the branch to execute on",
+        default="main",
+    )
+    tooling_branch_name: Optional[str] = Field(
+        description="The name of the branch to use for rerender+webservices \
+            github actions and conda-forge-ci-setup-feedstock references",
+        default="main",
+    )
+
+
+class GithubActionsConfig(BaseModel):
+    artifact_retention_days: Optional[int] = Field(
+        description="The number of days to retain artifacts",
+        default=14,
+    )
+
+    cancel_in_progress: Optional[bool] = Field(
+        description="Whether to cancel jobs in the same build if one fails.",
+        default=True,
+    )
+
+    max_parallel: Optional[Union[int, Nullable]] = Field(
+        description="The maximum number of jobs to run in parallel",
+        default=None,
+    )
+
+    self_hosted: Optional[bool] = Field(
+        description="Whether to use self-hosted runners",
+        default=False,
+    )
+
+    store_build_artifacts: Optional[bool] = Field(
+        description="Whether to store build artifacts",
+        default=False,
+    )
+
+    timeout_minutes: Optional[int] = Field(
+        default=360,
+        description="The maximum amount of time (in minutes) that a \
+            job can run before it is automatically canceled",
+    )
+
+    triggers: Optional[list] = Field(
+        default=[],
```

Comment:
```suggestion
        default_factory=list,
```

I think this is the canonical way of defining default mutables. Otherwise we get the same `function(value=[])` issue.

Reply:
No, see https://stackoverflow.com/a/63808835. Your suggestion results in `triggers: !!python/name:builtins.list ''` in `data/conda-forge.v2.yml`

Reply:
Oh nice, they have finally[ documented it](https://github.com/pydantic/pydantic/pull/6122/files). It always felt like an implementation detail so it felt weird to rely on it. Glad to see it's actually official.